spec/unit/shell/shell_session_spec.rb

```diff
@@ -180,68 +154,65 @@ def loading_complete
 
 describe Shell::SoloLegacySession do
   let(:json_attribs) { { "a" => "b" } }
+  let(:node) { Chef::Node.new }
+  let(:session) { Shell::SoloLegacySession.instance }
+  let(:recipe) { Chef::Recipe.new(nil, nil, run_context) }
+  let(:run_context) { Chef::RunContext.new(node, {}, Chef::EventDispatch::Dispatcher.new) }
 
   before do
     Chef::Config[:shell_config] = { override_runlist: [Chef::RunList::RunListItem.new("shell::override")] }
-    Chef::Config[:shell_solo] = true
-    @session = Shell::SoloLegacySession.instance
-    @node = Chef::Node.new
-    @events = Chef::EventDispatch::Dispatcher.new
-    @run_context = @session.run_context = Chef::RunContext.new(@node, {}, @events)
-    @session.node = @node
-    @recipe = @session.recipe = Chef::Recipe.new(nil, nil, @run_context)
-    Shell::Extensions.extend_context_recipe(@recipe)
-  end
-
-  after do
-    Chef::Config[:shell_solo] = nil
+    Chef::Config[:solo_legacy_shell] = true
```

Comment:
This is the meaningful change. This method determines which subclass of `Shell::Session` is used and it is looking for this key to be set in `Chef::Config`: https://github.com/chef/chef/blob/master/lib/chef/shell.rb#L189-L196

The string `shell_solo` is not present anywhere else in the project so this is likely some leftover cruft.

Reply:
yeah, this is some veeeery old code...